design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+First, the user will create an instance of a `ClusterStackInstall` that specifies which cluster stack they want to install.
+Then, when the Stack Manager installs the GCP stack, a few things will happen:
+
+* All CRDs related to GCP infrastructure and services will be installed to the cluster, such as `CloudSQLInstance`, `CloudSQLInstanceClass`, `GKECluster`, `GKEClusterClass`, `Network`, `Subnetwork`, etc.
+* A `ClusterRole` will be created at the cluster level that includes the RBAC rules for the [allowed core Kubernetes kinds](#allowed-resource-access), the stack's owned CRDs, and its dependent CRDs.
+* A `ServiceAccount` is created in the namespace that the stack's controller will be running.  This will serve as the runtime identity of the controller.
+* A `ClusterRoleBinding` is created at the cluster level to bind the `ClusterRole` and the namespaced `ServiceAccount`
+* A `Deployment` is created in the same namespace as the stack and it is expected to launch and manage the pod that hosts the GCP controllers.
+
+Because the stack is given a `ClusterRole` and `ClusterRoleBinding`, its permissions to the allowed resources are across all namespaces.
+
+### Architecture Diagram - Cluster Stack
+
+The diagram below summarizes the artifacts in the control plane after the GCP stack has been installed and a couple example `MySQLInstance` claims have been created.
+
+![Diagram of a Cluster Stack Installed](./images/cluster-stack-installed.png)
+
+### Environments
+
+While cluster stacks are installed at the cluster level (there is only one instance of each cluster stack), they can still support the concept of multiple "environments", each represented by a namespace.
+For example, an admin can create a namespace to model their production "environment" and another namespace to model their development "environment".
+All resources within each of those environments can share a common set of provider credentials.
+This namespaced "environment" approach provides the administrator with a way to collect or aggregate all the infrastructure and services associated with a given environment.
+
+#### Cluster Stack Single Instance Version Limitation
+
+One caveat of this approach is that because there is only a single instance of the cluster stack in the cluster, it is therefore shared across all environments.
+This means that all environments must use the same version of the stack.
+If an admin wants to try a newer version of the stack for their dev environment, it will also result in the newer version of the stack being used in the production environment as well.
+
+This limitation could be addressed in the future by allowing multiple versions of cluster stacks installed per control plane, which would require additional complexity and design work.
+To get started, we will make the simplifying constraint that only a single version of a cluster stack can be installed and it will be shared across all namespaces (environments).
+
+#### Environment Policies
```

Comment:
It seems like we're converging on this idea of managed resources being logically grouped by their Kubernetes namespace. We should be sure to update our documentation (which deals extensively in the `crossplane-system` namespace) to reflect this.

Reply:
Good point.  This thinking may be a bit forward looking, as our v0.3 user guides may not address or go into any depth on the concept of multiple environments such as "dev" or "prod".  We should consider that though if it makes sense in the user guide scenarios we want to focus on.